### PR TITLE
PS-2866 More strict check if the component should have access to MLflow

### PR DESCRIPTION
**Pull request by joker806**
https://keboola.atlassian.net/browse/PS-2866

More strict check if the component should have access to MLflow:
- component has to have a mlflow-artifacts-access feature
- stack or project has to have a sandboxes-python-mlflow
- project must exist on sandboxes-api

### CODE REVIEW

**Review by odinuv — APPROVED** [review]
seems legit 🤞

**Review by pivnicek — APPROVED** [review]
I don't see any logical issues but I wonder if it wouldn't have been enough to just catch the 404 in the end 😄 .  But this seems solid to me 👍

**Comment by odinuv** [comment]
That's something for discussion possibly - can it return a 403? now? in the future? Is is a reason good enough to crash all jobs? Or one might be extremely benevolent and simply say that failure to get the project (even 5xx) is logged as warning and simply means that the mlflow is not available in the job

**Comment by joker806** [comment]
In theory it's enought - ask sandboxes API, if it returns project with mlflow credentials, we can pass them. But the it would make such API call before each single job run. Adding those extra checks we can eliminate the API call.

**Comment by odinuv** [comment]
yes, but we're talking about what happens when the API call is made and fails

**Comment by odinuv** [comment]
come to think about this - i'd go with the extremeley benevolent way - only a fraction of transformations will use mlflow and in the worst case of sandbox API breakdown - this will minimize the impact

**Comment by joker806** [comment]
Ok, what about this? Warning is logged on any sandboxes API error. I removed the the 404 handling - I expect if the project/stack has the MLflow feature, the project is registered to sandboxes API.

File: src/Docker/MlflowProjectResolver.php [hunk 1]
```diff
@@ -45,11 +51,13 @@ public function getMlflowProjectIfAvailable(Component $component, array $tokenIn
...
-            }
-
-            throw $e;
+            $this->logger->warning(
+                'Failed to fetch MLflow project details. Access to MLflow artifacts will not be enabled!',
+                [
+                    'exception' => $e,
+                ]
+            );
+            return null;
```
Comment (pivnicek) [inline src/Docker/MlflowProjectResolver.php:60]: I like it 👍 .  We shouldn't let this side business fail the job,  if there's an mlflow issue they'll find out about it when they try to use it.

**Review by pivnicek — APPROVED** [review]

**Review by odinuv — APPROVED** [review]

**Pull request merged**